### ISSUE TITLE
cat /dev/urandom never closes

### ISSUE THREAD

**Issue by Melkor333** [cantino/mcfly#32]
I'm trying to build a package for Nixos, but when I source the macfly.bash file, it hangs on the following line:

https://github.com/cantino/mcfly/blob/a548e310152ec036a06625ba5521d51ad0ddd25c/mcfly.bash#L8

It's maybe due to the problem described [in this forum](https://www.unix.com/unix-for-advanced-and-expert-users/271649-cmd-cat-dev-urandom-not-closing-cleanly.html). Cat is never stopped...

What seems to stop the command is using `head -c 24` instead of the `fold ... | head -n 1`:

```
[sam@Waluigi:~/projects/mcfly]$ x=$(cat /dev/urandom | env LC_ALL=C tr -dc 'a-zA-Z0-9' | head -c 24)
tr: write error: Broken pipe
tr: write error
cat: write error: Broken pipe
[sam@Waluigi:~/projects/mcfly]$ echo $x
ktgQD1V2qVWbvuLUnFTEhwGg
```

But then the broken pipe errors appear as you can see. I think they could be discarded with
```
x=$(cat /dev/urandom 2>/dev/null| env LC_ALL=C tr -dc 'a-zA-Z0-9' 2>/dev/null | head -c 24)
```

**Comment by cantino** [comment]
Thanks @Melkor333! Your last snippet works for me on Mac and Linux. However, I'm a bit nervous about silencing all errors to `/dev/null`. Does `x=$(cat /dev/urandom | env LC_ALL=C tr -dc 'a-zA-Z0-9'  | head -c 24)` work for you?

**Comment by Melkor333** [comment]
Okay this is very strange. When I just tried the command `.... head -c 24` again without silencing errors, no errors appeared whatsoever. Even the original command `fold -w 24 | head -n 1` works now. So I think the PR doesn't *have* to be merged.
But I would like if you merged it anyways, as it wouldn't end in all new teminals being broken and instead just these `write error: Broken pip` appearing, if this situation ever occurs again. It's also a shorter line using a command less, so I guess nothing speaks against it :)

**Comment by cantino** [comment]
:+1:

**Comment by nfultz** [comment]
I was getting this issue on 0.3.3, fixed it here: https://github.com/nfultz/mcfly/pull/1/commits/b3189ccc4d61910315406bc76c7f1da79fc1ed79

**Comment by karsibali** [comment]
I experienced this problem with two different programs and I opened new issues within their github pages.
I had the same problem if I use mcfly with visual studio code. Here is the isseu link:
[https://github.com/microsoft/vscode-cpptools/issues/5275](url)
McFly also caused the same problem when I used it with hyper terminal:
[https://github.com/zeit/hyper/issues/4465](url)

When I disable mcfly everything works fine. The interesting thing is I do not have any problems if I use mcfly with gnome-terminal.
